packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +648,843 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+          const trailingText = currentOpenTag.content.substring(startOfRemainingText);
+          plainTextRepresentation += trailingText;
+        }
+
+        currentOpenTag.plainTextEndIndex = plainTextRepresentation.length;
+        addTag(currentOpenTag, tagParseStack, tags);
+      } else {
+        console.error(
+          'Unexpected close tag found. Got "' +
+            closeTagType +
+            '" but expected "' +
+            tagParseStack[tagParseStack.length - 1]?.tagType +
+            '"'
+        );
+      }
+    }
+
+    // Update parsing index; move past the end of the close tag
+    parseIndex = foundHtmlTag.startIdx + foundHtmlTag.content.length;
+  } // While parseIndex < text.length loop
+
+  return [tags, plainTextRepresentation];
+};
+
+/* @conditional-compile-remove(mention) */
+const parseOpenTag = (tag: string, startIdx: number): TagData => {
+  const tagType = tag
+    .substring(1, tag.length - 1)
+    .split(' ')[0]
+    .toLowerCase()
+    .replace('/', '');
+  return {
+    tagType,
+    openTagIdx: startIdx,
+    openTagBody: tag
+  };
+};
+
+/* @conditional-compile-remove(mention) */
+const findNextHtmlTag = (text: string, startIndex: number): HtmlTag | undefined => {
+  const tagStartIndex = text.indexOf('<', startIndex);
+  if (tagStartIndex === -1) {
+    // No more tags
+    return undefined;
+  }
+  const tagEndIndex = text.indexOf('>', tagStartIndex);
+  if (tagEndIndex === -1) {
+    // No close tag
+    return undefined;
+  }
+  const tag = text.substring(tagStartIndex, tagEndIndex + 1);
+  let type: HtmlTagType = 'open';
+  if (tag[1] === '/') {
+    type = 'close';
+  } else if (tag[tag.length - 2] === '/') {
+    type = 'self-closing';
+  }
+  return {
+    content: tag,
+    startIdx: tagStartIndex,
+    type
+  };
+};
+
+/* @conditional-compile-remove(mention) */
+const addTag = (tag: TagData, parseStack: TagData[], tags: TagData[]): void => {
+  // Add as sub-tag to the parent stack tag, if there is one
+  const parentTag = parseStack[parseStack.length - 1];
+
+  if (parentTag) {
+    // Adjust the open tag index to be relative to the parent tag
+    const parentContentStartIdx = parentTag.openTagIdx + parentTag.openTagBody.length;
+    const relativeIdx = tag.openTagIdx - parentContentStartIdx;
+    tag.openTagIdx = relativeIdx;
+  }
+
+  if (!tag.closeTagIdx) {
+    // If the tag is self-closing, the close tag is the same as the open tag
+    if (tag.openTagBody[tag.openTagBody.length - 2] === '/') {
+      tag.closeTagIdx = tag.openTagIdx;
+    } else {
+      // Otherwise, the close tag index is the open tag index + the open tag body + the content length
+      tag.closeTagIdx = tag.openTagIdx + tag.openTagBody.length + (tag.content ?? []).length;
+    }
+  }
+
+  // Put the tag where it belongs
+  if (!parentTag) {
+    tags.push(tag);
+  } else {
+    if (!parentTag.subTags) {
+      parentTag.subTags = [tag];
+    } else {
+      parentTag.subTags.push(tag);
+    }
+  }
+};
```

Comment:
Let's open a util file or any new file for those functions... this tsx file is just simply too long to read

Reply:
A new refactored task will be created to handle this functionality